language/move-analyzer/editors/code/src/context.ts

```diff
@@ -60,7 +67,7 @@ export class Context {
      * To read more about the messages sent and responses received by this client, such as
      * "initialize," read [the Language Server Protocol specification](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#initialize).
      **/
-    startClient(): void {
+    async startClient(): Promise<void> {
```

Comment:
Or because ci-test needs to wait for the symbol table parsing to be completed synchronously, which requires that the activation of the plugin is also synchronous

Reply:
> Or because ci-test needs to wait for the symbol table parsing to be completed synchronously, which requires that the activation of the plugin is also synchronous

Well, if "the activation of the plugin is also synchronous" then why do we need `async`.

More generally, I am not a TS expert, and that's why I asked one of colleagues to look at the TS part of this PR, but still - a few more comments in code explaining what the additions to the TS code do would help (also with future maintenance).